service/csvService.go

```diff
@@ -6,133 +6,219 @@ import (
 	"errors"
 	"fmt"
 	"io/ioutil"
-	"log"
 	"net/http"
 	"os"
+	"strconv"
+	"strings"
+
+	"github.com/oscarSantoyo/academy-go-q12021/model"
 
 	"github.com/fatih/structs"
+	"github.com/golobby/container"
+	"github.com/labstack/gommon/log"
 )
 
+// CsvService is used as an interface for this service.
 type CsvService interface {
-	FilterById(string) ([]Doc, error)
-	DownloadCsvData() error
+	FilterByID(string) ([]model.Doc, error)
+	DownloadCsvData(string) error
+	SearchByConditions(map[string]string) ([]model.Doc, error)
 }
```

Comment:
Hey Rollo I didn't get this comment, this is wrapping the responsibility of CSV operations as a service so it can be used across the system, moving this to the controller would mix stuff.